src/pocketmine/Player.php

```diff
@@ -2468,7 +2469,23 @@ public function handleInventoryTransaction(InventoryTransactionPacket $packet) :
 
 				switch($type){
 					case InventoryTransactionPacket::USE_ITEM_ON_ENTITY_ACTION_INTERACT:
-						break; //TODO
+						$item = $this->inventory->getItemInHand();
+						$clickPos = $packet->trData->clickPos;
+						$slot = $packet->trData->hotbarSlot;
```

Comment:
I don't think this makes sense. You're already assuming that the item used for this action is the currently held item, so I don't see why it's necessary to throw the slot around.